components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/impl/RegisterApiServiceImpl.java

```diff
@@ -41,13 +44,26 @@
     public Response registerPost(RegistrationDTO body, String contentType) throws NotFoundException {
 
         try {
-            APIMgtAdminService adminService = RestApiUtil.getAPIMgtAdminService();
-            String overwriteLabels = body.getLabelInfo().getOverwriteLabels();
-            List<Label> labels = MappingUtil.convertToLabels(body.getLabelInfo().getLabelList());
-            adminService.registerGatewayLabels(labels, overwriteLabels);
-            //TODO : Add registration summary details based on the sharing values
-            RegistrationSummaryDTO registrationSummaryDTO = new RegistrationSummaryDTO();
-            return Response.ok().entity(registrationSummaryDTO).build();
+            LabelInfoDTO labelInfoDTO = body.getLabelInfo();
+
+            if (labelInfoDTO != null) {
+                APIMgtAdminService adminService = RestApiUtil.getAPIMgtAdminService();
+                String overwriteLabels = labelInfoDTO.getOverwriteLabels();
+                List<Label> labels = MappingUtil.convertToLabels(labelInfoDTO.getLabelList());
+                adminService.registerGatewayLabels(labels, overwriteLabels);
+                //TODO : Add registration summary details based on the sharing values
+                RegistrationSummaryDTO registrationSummaryDTO = new RegistrationSummaryDTO();
+                return Response.ok().entity(registrationSummaryDTO).build();
```

Comment:
Yes, this is a post request. This is a registration API which gateways will use at the startup. First start of the gateway, it will create the resource if that is not available at the core. Also, it will not create the resource on the next restarts. If we send 201, then we need to send the location header as well. If we send the location header, this registration details should be available with the GET request.